recipe/patches/0001-Redis-deps-now-build-but-do-not-link.patch

```diff
@@ -14,41 +14,6 @@ Signed-off-by: H. Vetinari <h.vetinari@gmx.com>
  3 files changed, 81 insertions(+), 13 deletions(-)
  rename thirdparty/patches/{redis-quiet.patch => 0001-keep-redis-quiet-and-override-AR.patch} (53%)
 
-diff --git a/bazel/BUILD.redis b/bazel/BUILD.redis
-index 9edccf5780..60e070028f 100644
---- a/bazel/BUILD.redis
-+++ b/bazel/BUILD.redis
-@@ -42,13 +42,17 @@ make(
- 
- genrule_cmd = select({
-     "@bazel_tools//src/conditions:darwin": """
--        unset CC LDFLAGS CXX CXXFLAGS
-+        export CC=$(CC)
-+        export CFLAGS=$(CC_FLAGS)
-+        export AR=$${CC/gnu-cc/gnu-ar}
-+        export NM=$${CC/gnu-cc/gnu-nm}
-+        export RANLIB=$${CC/gnu-cc/gnu-ranlib}
-         tmpdir="redis.tmp"
-         p=$(location Makefile)
-         cp -p -L -R -- "$${p%/*}" "$${tmpdir}"
-         chmod +x "$${tmpdir}"/deps/jemalloc/configure
-         parallel="$$(getconf _NPROCESSORS_ONLN || echo 1)"
--        make -s -C "$${tmpdir}" -j"$${parallel}" V=0 CFLAGS="$${CFLAGS-} -DLUA_USE_MKSTEMP -Wno-pragmas -Wno-empty-body"
-+        make -s -C "$${tmpdir}" -j"$${parallel}" V=0 CFLAGS="$${CFLAGS-} -DLUA_USE_MKSTEMP -Wno-pragmas -Wno-empty-body" AR="$${AR}" RANLIB="$${RANLIB}"
-         mv "$${tmpdir}"/src/redis-server $(location redis-server)
-         chmod +x $(location redis-server)
-         mv "$${tmpdir}"/src/redis-cli $(location redis-cli)
-@@ -76,5 +80,10 @@ genrule(
-     ],
-     cmd = genrule_cmd,
-     visibility = ["//visibility:public"],
-+    toolchains = [
-+        "@bazel_tools//tools/cpp:current_cc_toolchain",
-+        "@bazel_tools//tools/cpp:current_cc_host_toolchain",
-+        "@bazel_tools//tools/cpp:cc_flags",
-+    ],
```

Comment:
If you remove this part of the changes, does the build still pass?

Reply:
The build fails because `$(CC)` is not defined due to the sandboxing. We could in principle set `--action_env=CC` (and same for the other flags) rather than pass the conda compiler as `$(CONDA_CC)` so it will be defined, but I wanted to avoid accidentally overriding any other config that ray is doing by settings those variables just to get the conda compiler to redis.